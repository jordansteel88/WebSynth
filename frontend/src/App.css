.App {
  text-align: center;
  width: 100%;
  align-items: center;
  justify-content: center;
}

.App-header {
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #F7F7F7;
}


