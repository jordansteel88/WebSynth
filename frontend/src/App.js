// import React, { useState, useEffect, useMemo, useCallback } from 'react';
import React, { useEffect} from 'react';
import './App.css';
import { attack, release } from './api/api';
import Keyboard from './components/Keyboard'
// import KeyboardContext from './KeyboardContext';

function App() {
  useEffect(() => {
    const handleKeydown = (evt) => {
      if (evt.repeat) {
        return;
      }

      console.log(evt.key.toUpperCase());
      let target = document.getElementById(evt.key.toUpperCase())
      if (target) target.classList += ' pressed';

      attack(evt);
    };

    window.addEventListener('keydown', handleKeydown);
    return () => {
      window.removeEventListener('keydown', handleKeydown)
    }
  }, []);

  useEffect(() => {
    const handleKeyup = (evt) => {
      let target = document.getElementById(evt.key.toUpperCase())
      if (target) target.classList.remove('pressed');

      release(evt);
    };

    window.addEventListener('keyup', handleKeyup);
    return () => {
      window.removeEventListener('keyup', handleKeyup);
    }
  }, []);

  return (
      <div className="App">
        <div className="App-header">
          <Keyboard />  
        </div>
      </div>
  )
}


export default App;
