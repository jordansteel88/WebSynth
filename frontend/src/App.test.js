import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { BrowserRouter } from "react-router-dom";
import UserContext from './auth/UserContext';

jest.mock('./components/Synth.js', () => {
  return function DummySynth() {
    return (
      <div>DummySynth</div>
  )};
})

describe("App", () => {
  const initialState = {
    currentUser: "test"
  }

  it("renders without crashing", function() {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <App />
        </UserContext.Provider> 
      </BrowserRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <App />
        </UserContext.Provider> 
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
})
