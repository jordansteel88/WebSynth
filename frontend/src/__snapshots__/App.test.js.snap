// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Navigation"
    >
      <nav
        class="navbar navbar-expand-md"
      >
        <ul
          class="nav"
        >
          <li
            class="mr-3 nav-item"
          >
            <a
              aria-current="page"
              class="text-white active"
              href="/"
            >
              WebSynth
            </a>
          </li>
        </ul>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="ml-auto nav"
          >
            <li
              class="mr-3 nav-item"
            >
              <a
                class="text-white"
                href="/login"
              >
                Log In
              </a>
            </li>
            <li
              class="mr-2 nav-item"
            >
              <a
                class="text-white"
                href="/register"
              >
                Sign Up
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div
      class="App-header"
    >
      <div>
        <div
          class="Home"
        >
          <div
            class="EffectsPanel"
          >
            <div>
              DummySynth
            </div>
            <h2>
              Log in or sign up to access more features!
            </h2>
          </div>
          <div
            class="Keyboard"
          >
            <div
              class="Key"
              id="CAPSLOCK"
            >
              <div
                class="Key-text capsLock"
              >
                CAPSLOCK
              </div>
            </div>
            <div
              class="Key-flat"
              id="Q"
            >
              <div
                class="Key-text u-white"
              >
                Q
              </div>
            </div>
            <div
              class="Key"
              id="A"
            >
              <div
                class="Key-text"
              >
                A
              </div>
            </div>
            <div
              class="Key-flat"
              id="W"
            >
              <div
                class="Key-text u-white"
              >
                W
              </div>
            </div>
            <div
              class="Key"
              id="S"
            >
              <div
                class="Key-text"
              >
                S
              </div>
            </div>
            <div
              class="Key-flat"
              id="E"
            >
              <div
                class="Key-text u-white"
              >
                E
              </div>
            </div>
            <div
              class="Key"
              id="D"
            >
              <div
                class="Key-text"
              >
                D
              </div>
            </div>
            <div
              class="Key"
              id="F"
            >
              <div
                class="Key-text"
              >
                F
              </div>
            </div>
            <div
              class="Key-flat"
              id="T"
            >
              <div
                class="Key-text u-white"
              >
                T
              </div>
            </div>
            <div
              class="Key"
              id="G"
            >
              <div
                class="Key-text"
              >
                G
              </div>
            </div>
            <div
              class="Key-flat"
              id="Y"
            >
              <div
                class="Key-text u-white"
              >
                Y
              </div>
            </div>
            <div
              class="Key"
              id="H"
            >
              <div
                class="Key-text"
              >
                H
              </div>
            </div>
            <div
              class="Key"
              id="J"
            >
              <div
                class="Key-text"
              >
                J
              </div>
            </div>
            <div
              class="Key-flat"
              id="I"
            >
              <div
                class="Key-text u-white"
              >
                I
              </div>
            </div>
            <div
              class="Key"
              id="K"
            >
              <div
                class="Key-text"
              >
                K
              </div>
            </div>
            <div
              class="Key-flat"
              id="O"
            >
              <div
                class="Key-text u-white"
              >
                O
              </div>
            </div>
            <div
              class="Key"
              id="L"
            >
              <div
                class="Key-text"
              >
                L
              </div>
            </div>
            <div
              class="Key-flat"
              id="P"
            >
              <div
                class="Key-text u-white"
              >
                P
              </div>
            </div>
            <div
              class="Key"
              id=";"
            >
              <div
                class="Key-text"
              >
                ;
              </div>
            </div>
            <div
              class="Key"
              id="'"
            >
              <div
                class="Key-text"
              >
                '
              </div>
            </div>
          </div>
          <div
            class="SaveForm-container"
          >
            <form
              class="SaveForm mt-5 w-50 d-inline"
            >
              <div
                class="input-group mt-5 mx-auto w-50"
              >
                <input
                  aria-describedby="button-save-profile"
                  aria-label="save profile"
                  class="form-control SaveForm-input form-control"
                  id="name"
                  name="name"
                  placeholder="Enter a name for your profile..."
                  type="text"
                  value=""
                />
                <button
                  class="btn btn btn-primary"
                  id="button-save-profile"
                  type="button"
                >
                  Save Profile
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
